go/vt/vtgate/executor_test.go

```diff
@@ -1869,6 +1869,64 @@ func TestGenerateCharsetRows(t *testing.T) {
 	}
 }
 
+func TestExecutorMaxPayloadSizeExceeded(t *testing.T) {
+	saveMax := *maxPayloadSize
+	saveWarn := *warnPayloadSize
+	*maxPayloadSize = 10
+	*warnPayloadSize = 5
+	defer func() {
+		*maxPayloadSize = saveMax
+		*warnPayloadSize = saveWarn
+	}()
+
+	executor, _, _, _ := createExecutorEnv()
+	session := NewSafeSession(&vtgatepb.Session{TargetString: "@master"})
+	warningCount := warnings.Counts()["PayloadSizeExceeded"]
+	testMaxPayloadSizeExceeded := []string{
+		"select * from main1",
+		"insert into main1(id) values (1), (2)",
+		"update main1 set id=1",
+		"delete from main1 where id=1",
+	}
+	for _, query := range testMaxPayloadSizeExceeded {
+		_, err := executor.Execute(context.Background(), "TestExecutorMaxPayloadSizeExceeded", session, query, nil)
+		want := "query payload size above threshold"
+		if err == nil || err.Error() != want {
+			t.Errorf("got: %v, want %s", err, want)
+		}
+	}
+	if got, want := warnings.Counts()["PayloadSizeExceeded"], warningCount; got != want {
```

Comment:
replace `t.Errorf` with `assert.Equal`

Reply:
Updated to use `assert.Equal` wherever possible